src/libs/ModifiedExpenseMessage.ts

```diff
@@ -188,16 +186,31 @@ function getForReportAction(reportID: string | undefined, reportAction: OnyxEntr
 
     const hasModifiedTag = reportActionOriginalMessage && 'oldTag' in reportActionOriginalMessage && 'tag' in reportActionOriginalMessage;
     if (hasModifiedTag) {
-        buildMessageFragmentForValue(
-            PolicyUtils.getCleanedTagName(reportActionOriginalMessage?.tag ?? ''),
-            PolicyUtils.getCleanedTagName(reportActionOriginalMessage?.oldTag ?? ''),
-            policyTagListName,
-            true,
-            setFragments,
-            removalFragments,
-            changeFragments,
-            policyTagListName === Localize.translateLocal('common.tag'),
-        );
+        const policyTags = allPolicyTags?.[`${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`] ?? {};
+        const transactionTag = reportActionOriginalMessage?.tag ?? '';
+        const oldTransactionTag = reportActionOriginalMessage?.oldTag ?? '';
+        const splittedTag = transactionTag.split(CONST.COLON);
+        const splittedOldTag = oldTransactionTag.split(CONST.COLON);
+
+        Object.keys(policyTags).forEach((policyTagKey, index) => {
+            const policyTagListName = PolicyUtils.getTagListName(policyTags, index) || Localize.translateLocal('common.tag');
```

Comment:
Isn't `??` preferred over `||`?

Reply:
Yes, it is. However, for this case there is a case when the 1st method returns an empty string, then we should use a fallback string from translations.